spec/fluent.asdl

```diff
@@ -31,9 +31,9 @@ module Fluent
           | Junk(str content)
           attributes (annot* annotations, span? span)
 
-    base_message = Message(pat? value, attr* attributes)
-                 | PrivateMessage(pat value, tag* tags)
-                 attributes (iden id, comment? comment)
+    base_message = Message
+                 | PrivateMessage
+                 attributes (iden id, pat? value, attr* attributes, comment? comment)
```

Comment:
#68 made `value` required on `PrivateMessage`. That's because a message consisting only of tags doesn't make much sense. With tags removed, I still think it's unlikely to have private messages consisting only of attributes. Allowing it, however, makes private messages consistent with public messages. It reduces the number of special rules to remember.

Reply:
I'd prefer to keep the requirement for private messages to have a value, otherwise programmers will abuse the uncontrollable hashmap of a private public attribute thing.

Reply:
Fair point. We can enforce it in the EBNF and keep the ASDL simple.